packages/@lightningjs/ui-components/src/mixins/withUpdates/index.js

```diff
@@ -40,7 +40,18 @@ function getPropertyDescriptor(name, key) {
         if (changeHandler && typeof changeHandler === 'function') {
           value = changeHandler.call(this, value);
         }
-        this[key] = key === 'style' ? clone(this[key], value) : value;
+        const newValue = key === 'style' ? clone(this[key], value) : value;
+
+        if (
+          typeof this[key] === 'object' &&
+          this[key] !== null &&
+          this[key].style
+        ) {
+          const style = clone(this[key].style, value.style || {});
+          newValue.style = style;
```

Comment:
can you throw in a comment to this block? Maybe something like: "if this property passed in is meant to be used in a nested component, deep merge the style with any existing style properties on the nested component (like from a componentConfig)"